server/src/internalClusterTest/java/org/elasticsearch/repositories/InvalidRepositoryIT.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    public static class UnstableRepository extends MockRepository {
+        public static final String TYPE = "unstable";
+        public static final Setting<List<String>> UNSTABLE_NODES = Setting.stringListSetting(
+            "repository.unstable_nodes",
+            Setting.Property.NodeScope,
+            Setting.Property.Dynamic
+        );
+
+        public UnstableRepository(
+            RepositoryMetadata metadata,
+            Environment environment,
+            NamedXContentRegistry namedXContentRegistry,
+            ClusterService clusterService,
+            BigArrays bigArrays,
+            RecoverySettings recoverySettings
+        ) {
+            super(metadata, environment, namedXContentRegistry, clusterService, bigArrays, recoverySettings);
+            List<String> unstableNodes = UNSTABLE_NODES.get(metadata.settings());
+            if (unstableNodes.contains(clusterService.getNodeName())) {
+                throw new RepositoryException(metadata.name(), "Failed to create repository: current node is not stable");
+            }
+        }
+
+        public static class Plugin extends org.elasticsearch.plugins.Plugin implements RepositoryPlugin {
+            @Override
+            public Map<String, Factory> getRepositories(
+                Environment env,
+                NamedXContentRegistry namedXContentRegistry,
+                ClusterService clusterService,
+                BigArrays bigArrays,
+                RecoverySettings recoverySettings
+            ) {
+                return Collections.singletonMap(
+                    TYPE,
+                    (metadata) -> new UnstableRepository(metadata, env, namedXContentRegistry, clusterService, bigArrays, recoverySettings)
+                );
+            }
+
+            @Override
+            public List<Setting<?>> getSettings() {
+                return List.of(UNSTABLE_NODES);
+            }
+        }
+    }
+
+    public void testCreateInvalidRepository() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String repositoryName = "test-duplicate-create-repo";
+
+        // put repository for the first time: only let master node create repository successfully
+        createRepository(
+            repositoryName,
+            UnstableRepository.TYPE,
+            Settings.builder()
+                .put("location", randomRepoPath())
+                .putList(
+                    UnstableRepository.UNSTABLE_NODES.getKey(),
+                    Arrays.stream(internalCluster().getNodeNames())
+                        .filter(name -> name.equals(internalCluster().getMasterName()) == false)
+                        .toList()
+                )
+        );
+        // verification should fail with some node has InvalidRepository
+        boolean verifyPass = false;
+        try {
+            client().admin().cluster().prepareVerifyRepository(repositoryName).get();
```

Comment:
Mentioned before, but this should be `final var repositoryVerificationException = expectThrows(RepositoryVerificationException.class, () -> client().admin().cluster().prepareVerifyRepository(repositoryName).get());`

Note that `expectThrows` returns the exception it caught so you can assert things about it such as its causes and suppressed exceptions.